### PR TITLE
fix: chapter 10 feedback qa component fixes

### LINKED ISSUE #1169: Chapter 10 mobile layouts are broken
When [Christoph was doing chapter 10 QA](https://www.figma.com/design/LqjK3Tpvd9KJ4buFArCJBQ/Saving-Satoshi?node-id=9670-3299&t=ZY5cW5xINF2iLvRl-1) he noticed the mobile layouts are broken. As it is right now, chapter 10 is not usable on mobile.

Screenshot from 2024-11-10 13-48-03
Screenshot from 2024-11-10 13-48-16

### LINKED ISSUE #1179: Share to social button when chapter 10 is completed
We have no button to prompt the user to share that they completed chapter 10. This is the end of the game too so it's a special one! I know the idea of nostr badges has been floated around, but it is a separate ticket (https://github.com/saving-satoshi/saving-satoshi/issues/1141)

This feature request was initially raised by someone in our Discord channel: https://discord.com/channels/903125802726596648/1089879043626778655/1306594337894826077

Screenshot from 2024-11-14 15-29-47

### PR DESCRIPTION
**Pull request by benalleng**
Adds the component fixes for chapter 10 qa feedback
closes #1169 
closes #1179

### CODE REVIEW

File: utils/data.ts
Comment (satsie) [inline utils/data.ts:53]: I think this one stays as 101,000, right?

I sat down to review this PR then the baby woke up :sob:. Going to leave this single lonely comment here and try to continue my review tomorrow.

File: ui/lesson/ScriptingChallenge/Runner/index.tsx
Comment (0tuedon) [inline ui/lesson/ScriptingChallenge/Runner/index.tsx:181]: I assume, we want this log?

File: ui/lesson/TransactionsChallenge/index.tsx
Comment (0tuedon) [inline ui/lesson/TransactionsChallenge/index.tsx:271]: A niit here would be if we can hide the scroll or change the default scroll on windows os

File: ui/lesson/TransactionsChallenge/index.tsx
Comment (0tuedon) [inline ui/lesson/TransactionsChallenge/index.tsx:427]: Last challenge?

**Review by 0tuedon — CHANGES_REQUESTED** [review]
LGTM, Left some few questions and niits

File: ui/lesson/TransactionsChallenge/index.tsx
Comment (benalleng) [reply ui/lesson/TransactionsChallenge/index.tsx]: swapped this as we start with a prefilled page and then have one page at the very end

File: ui/lesson/TransactionsChallenge/index.tsx
Comment (benalleng) [reply ui/lesson/TransactionsChallenge/index.tsx:271]: you mean like just make sure it is scrolled to the end on render?

File: content/lessons/chapter-10/opening-a-channel-4.tsx
Comment (satsie) [inline content/lessons/chapter-10/opening-a-channel-4.tsx:28]: How does this answer script work if it is missing the `OP_2` elements required for `OP_CHECKMULTISIG`?

File: utils/data.ts
Comment (satsie) [inline utils/data.ts:72]: Does this code validate answers? If the input for the refund tx is 100,000 sats, and you take away 1,000 for miner fees, the output should be 99,000 sats

File: utils/data.ts
Comment (satsie) [inline utils/data.ts:87]: Same comment as above, I believe this should be 99,000 sats (related to https://github.com/saving-satoshi/saving-satoshi/issues/1175)

File: utils/data.ts
Comment (satsie) [inline utils/data.ts:103]: Same as the two comments above since these are all different versions of the same refund transaction :) Should be 99,000 sats

File: utils/data.ts
Comment (satsie) [inline utils/data.ts:135]: I haven't played through the last part of chapter 10 yet ("Making a payment"). I'm not 100% all these numbers are correct. Going to play through it now and circle back to review this last set of numbers.

Comment (benalleng) [reply utils/data.ts:135]: should we now say that you end up with 98,000 when paying the 1,000 sat to laszlo?

**Comment by satsie** [comment]
The mobile layouts seem like they have been greatly improved. Thanks for tackling this! I don't know if we care, but mobile in landscape/horizontal is still a bit buggy. On the right half of the screen, I can't scroll up to see the input/output part of the exercise. I'll defer to someone else to decide if this is actually important or not.

Screenshot from 2024-11-15 13-28-27
Screenshot from 2024-11-15 13-29-01

File: content/lessons/chapter-10/opening-a-channel-4.tsx
Comment (benalleng) [reply content/lessons/chapter-10/opening-a-channel-4.tsx:28]: It just runs a `.includes()` method on the user's script array so the `OP_CHECKMULTISIG` already requires those `OP_2` commands

Comment (satsie) [reply content/lessons/chapter-10/opening-a-channel-4.tsx:28]: oh, nice!

File: utils/data.ts
Comment (satsie) [reply utils/data.ts:135]: Here's a comment from MZ on this: https://github.com/saving-satoshi/saving-satoshi/pull/1174#discussion_r1842343476
Screenshot from 2024-11-14 13-37-26

Here's my understanding:
- You start with 101,000 sats
- The 2-of-2 multisig has `input`: 101,000 sats, `output`: 100,000 sats because 1,000 sats went to miner fees
- The commitment transaction should always have `input`: 100,000 sats, and the `output` should add up to 99,000 sats because we reserve 1,000 sats for fees
- When you buy a beer from Laszlo, the commitment transaction works out to be `input`: 100,000, `output`: 98,000 sats to you, and 1,000 sats to Laszlo. You most recent comment seems correct to me.

Comment (satsie) [reply utils/data.ts:135]: I reviewed the numbers you updated in 788751e65ddc028de04bf8d084fbd12a9c3f5890 and they look good! Resolving this comment.

File: utils/data.ts
Comment (satsie) [reply utils/data.ts]: fixed in https://github.com/saving-satoshi/saving-satoshi/commit/788751e65ddc028de04bf8d084fbd12a9c3f5890

File: utils/data.ts
Comment (satsie) [reply utils/data.ts]: fixed in https://github.com/saving-satoshi/saving-satoshi/commit/788751e65ddc028de04bf8d084fbd12a9c3f5890

**Review by satsie — APPROVED** [review]
Thanks for making all these updates! I'm approving but I acknowledge it still needs Tuedon's approval.

Just a note, this PR contains a number of the same fixes as one I've been working on: https://github.com/saving-satoshi/saving-satoshi/pull/1174
- change "multi-sig" to "multisig"
- fix the cost of the SHA-256 Stout
- fix a typo on one of the resource pages: `chapter_ten.resources.ouput_zero_sig`

Depending on who merges first, this will potentially cause conflicts for one of the PRs.

I am going to file the [horizontal mobile issue separately](https://github.com/saving-satoshi/saving-satoshi/issues/1183). We can even close it if we think it's not worth fixing.

**Comment by benalleng** [comment]
Yeah, I need to rebase those changes back from your PR so there aren't conflicts

**Comment by satsie** [comment]
I think we can wait to see which  gets merged to dev first 😄 1,000 sats it's going to be this PR 🤣🤣

**Pull request merged**
